Assets/SteamVR_Unity_Toolkit/Scripts/Controls/3D/VRTK_Door.cs

```diff
@@ -0,0 +1,365 @@
+                {
+                    direction = Direction.x;
+                }
+                else
+                {
+                    direction = Direction.y;
+                }
+            }
+
+            return direction;
+        }
+
+        private void InitFrame()
+        {
+            if (frame == null)
+            {
+                return;
+            }
+
+            frameRb = frame.GetComponent<Rigidbody>();
+            if (frameRb == null)
+            {
+                frameRb = frame.AddComponent<Rigidbody>();
+                frameRb.isKinematic = true; // otherwise frame moves/falls over when door is moved or fully open
+            }
+        }
+
+        private void InitDoor()
+        {
+            Utilities.CreateColliders(getDoor());
+
+            doorRb = getDoor().GetComponent<Rigidbody>();
+            if (doorRb == null)
+            {
+                doorRb = getDoor().AddComponent<Rigidbody>();
+                doorRb.angularDrag = 10;
+                doorRb.collisionDetectionMode = CollisionDetectionMode.Continuous; // otherwise door will not react to fast moving controller
+            }
+
+            doorHj = getDoor().GetComponent<HingeJoint>();
+            if (doorHj == null)
+            {
+                doorHj = getDoor().AddComponent<HingeJoint>();
+                doorHjCreated = true;
+            }
+            doorHj.connectedBody = frameRb;
+
+            doorCf = getDoor().GetComponent<ConstantForce>();
+            if (doorCf == null)
+            {
+                doorCf = getDoor().AddComponent<ConstantForce>();
+                doorCf.enabled = false;
+                doorCfCreated = true;
+            }
+        }
+
+        private void InitHandle()
+        {
+            if (handles == null)
+            {
+                return;
+            }
+
+            if (handles.GetComponentInChildren<Collider>() == null)
+            {
+                Utilities.CreateColliders(handles);
+            }
+
+            handleRb = handles.GetComponent<Rigidbody>();
+            if (handleRb == null)
+            {
+                handleRb = handles.AddComponent<Rigidbody>();
+            }
+            handleRb.isKinematic = false;
+            handleRb.useGravity = false;
+
+            handleFj = handles.GetComponent<FixedJoint>();
+            if (handleFj == null)
+            {
+                handleFj = handles.AddComponent<FixedJoint>();
+                handleFj.connectedBody = doorRb;
+            }
+
+            handleIo = handles.GetComponent<VRTK_InteractableObject>();
+            if (handleIo == null)
+            {
+                handleIo = handles.AddComponent<VRTK_InteractableObject>();
+            }
+            handleIo.isGrabbable = true;
+            handleIo.precisionSnap = true;
+            handleIo.grabAttachMechanic = VRTK_InteractableObject.GrabAttachType.Spring_Joint;
```

Comment:
try this instead:
`handleIo.grabAttachMechanic = VRTK_InteractableObject.GrabAttachType.Track_Object;`

I feel the `Track_Object` mechanic gives a better grab feel on the door. The rotator track may work even better, but I couldn't get it to work on the tiny door (under the chest)